cmsl1t/analyzers/HW_Emu_jetMet_rates.py

```diff
@@ -258,29 +260,52 @@ def make_plots(self):
                 'Error: Please specify thresholds in the config .yaml in dictionary format')
 
         # print hw vs emu for rates and rate vs pileup
-        if self._doEmu:
-            for histo_name in self._sumTypes + self._jetTypes:
+        for histo_name in self._sumTypes + self._jetTypes:
+            plotter = getattr(self, histo_name + '_rates')
+            if self._doEmu:
                 if "_Emu" in histo_name:
                     continue
-                plotter = getattr(self, histo_name + '_rates')
                 emu_plotter = getattr(self, histo_name + '_Emu_rates')
-                plotter.overlay_with_emu(emu_plotter)
+                plotter.overlay([emu_plotter])
+            if self._doComp:
+                plotter.overlay([getattr(other_analyzer, histo_name + '_rates')
+                                for other_analyzer in other_analyzers])
 
-                plotter = getattr(self, histo_name + '_rate_vs_pileup')
+            plotter = getattr(self, histo_name + '_rate_vs_pileup')
+            if self._doEmu:
                 emu_plotter = getattr(self, histo_name + "_Emu" + '_rate_vs_pileup')
-                plotter.overlay_with_emu(emu_plotter)
+                plotter.overlay([emu_plotter])
+            if self._doComp:
+                plotter.overlay([getattr(other_analyzer, histo_name + '_rate_vs_pileup')
+                                for other_analyzer in other_analyzers])
 
         # calculate cumulative histograms
-        if not self._doEmu:
-            for plot in self.all_plots:
-                if 'rate_vs_pileup' not in plot.filename_format:
-                    hist = plot.plots.get_bin_contents([bn.Base.everything])
-                    hist = cumulative_hist(hist)
-                    setattr(self, plot.online_name, hist)
+        for plot in self.all_plots:
+            if 'rate_vs_pileup' not in plot.filename_format:
+                hist = plot.plots.get_bin_contents([bn.Base.everything])
+                hist = cumulative_hist(hist)
+                setattr(self, plot.online_name, hist)
+                if not self._doEmu:
                     plot.draw()
 
         print('  thresholds:')
 
+        if self.menuRates:
+            for histo_name in self._sumTypes + self._jetTypes:
+                h = getattr(self, histo_name)
+                h = normalise_to_collision_rate(h)
+                targetRate = self.menuRates.get(histo_name.replace('_Emu', ''))
+                threshold = None
+                closestRateDiff = 999999999
+                threshold = None
+                for i in range(h.nbins()):
```

Comment:
a more pythonic way would be to use something like
```python
for bin_content, low_bin_edge in zip(h, h.xedges()):
```
assuming `h` returns content for all bins and `h.xedges()` a list of lower edges. Individual calls to `get_bin_low_edge` and `get_bin_content` will be slower